scrapyd/app.py

```diff
@@ -33,15 +38,22 @@ def application(config):
     launcher = laucls(config, app)
 
     webpath = config.get('webroot', 'scrapyd.website.Root')
-    webcls = load_object(webpath)
+    website = server.Site(load_object(webpath)(config, app))
 
     timer = TimerService(poll_interval, poller.poll)
-    webservice = TCPServer(http_port, server.Site(webcls(config, app)), interface=bind_address)
-    log.msg(format="Scrapyd web console available at http://%(bind_address)s:%(http_port)s/",
-            bind_address=bind_address, http_port=http_port)
 
     launcher.setServiceParent(app)
     timer.setServiceParent(app)
-    webservice.setServiceParent(app)
+
+    if http_port:
+        webservice = TCPServer(http_port, website, interface=bind_address)
+        log.msg(format="Scrapyd web console available at http://%(bind_address)s:%(http_port)s/",
+                http_port=http_port, bind_address=bind_address)
+        webservice.setServiceParent(app)
+    if uds_path:
+        webservice = UNIXServer(uds_path, website, mode=0o660)
+        log.msg(format=u"Scrapyd web console available at http+unix://%(uds_path)s",
+                uds_path=uds_path)
+        webservice.setServiceParent(app)
```

Comment:
I'm not certain if I understand the logic here, is this really meant to start both TCPServer and UnixServer? Or is there some intention to allow to disable and stop TCPServer by setting http_port to zero?

Because with this implementation if someone just adds uds_path without removing http_port it will start both of them. Is that expected?